src/Routing/Route/Route.php

```diff
@@ -701,7 +701,10 @@ public function match(array $url, array $context = []): ?string
         unset($url['_method'], $url['[method]'], $defaults['_method']);
 
         // Defaults with different values are a fail.
-        if (array_intersect_key($url, $defaults) != $defaults) {
+        // Filter out null values from defaults for comparison as null values
+        // should be treated as "not set" for matching purposes
+        $filteredDefaults = array_filter($defaults, fn($v) => $v !== null);
```

Comment:
Should we filter defaults when the route is constructed? Doing the filter here will be more expensive as this is the hot-path of route matching.

Reply:
The updated change moves the null filtering logic for plugin and prefix keys from using array_filter() on every match() call to checking these specific keys inline during the
  matching loop. This addresses the performance concern raised by avoiding the overhead of creating a filtered array in the hot-path of route matching.
  
 Moving to construct doesnt seem possible, but correct me If I am wrong.